app/controllers/shops_controller.rb

```diff
@@ -37,6 +37,10 @@ def update
  	end
 
  	def destroy
+		@shop = Shop.find(params[:id])
```

Comment:
Use `current_user.shops.find...` to only allow permission to delete their own shop. Current code allows anyone to delete anyones shop. Use `current_user` relation for now as we don't have an authorisation system yet.